_unittest/test_21_Circuit.py

```diff
@@ -147,49 +148,26 @@ def test_12_connect_components(self):
         mycap = self.aedtapp.modeler.schematic.create_capacitor("C100", 1e-12)
         portname = self.aedtapp.modeler.schematic.create_interface_port("Port1")
         assert "Port1" in portname.name
-
-        assert self.aedtapp.modeler.connect_schematic_components(myind.id, myind.id, pinnum_second=2)
+        assert myind.pins[0].connect_to_component(portname.pins[0])
+        assert myind.pins[1].connect_to_component(myres.pins[1])
         assert self.aedtapp.modeler.connect_schematic_components(myres.id, mycap.id, pinnum_first=1)
-
+        gnd = self.aedtapp.modeler.schematic.create_gnd()
+        assert mycap.pins[1].connect_to_component(gnd.pins[0])
         # create_interface_port
         L1_pins = myind.pins
         L1_pin2location = {}
         for pin in L1_pins:
             L1_pin2location[pin.name] = pin.location
 
-        C1_pins = mycap.pins
-        C1_pin2location = {}
```

Comment:
@maxcapodi78 Is it expected to get rid of all those ports in the test?

Reply:
yes